amqp/transport.py

```diff
@@ -341,11 +341,16 @@ def _wrap_socket_sni(self, sock, keyfile=None, certfile=None,
                 opts['ssl_version'] = ssl.PROTOCOL_TLS
             else:
                 opts['ssl_version'] = ssl.PROTOCOL_SSLv23
+        sock = ssl.wrap_socket(**opts)
         # Set SNI headers if supported
         if (server_hostname is not None) and (
-                hasattr(ssl, 'HAS_SNI') and ssl.HAS_SNI):
-            opts['server_hostname'] = server_hostname
-        sock = ssl.SSLSocket(**opts)
+                hasattr(ssl, 'HAS_SNI') and ssl.HAS_SNI) and (
+                hasattr(ssl, 'SSLContext')):
+            context = ssl.SSLContext(opts['ssl_version'])
+            context.verify_mode = cert_reqs
+            context.check_hostname = True
+            context.load_cert_chain(certfile, keyfile)
```

Comment:
also if this was none this should not be executed

Reply:
Feel free to send a PR.